### ISSUE TITLE
Libtest produces invalid JSONs for doctests on Windows

### ISSUE THREAD

**Issue by mchernyavsky** [rust-lang/rust#76151]
Example project:
* `lib.rs`
```rust
/// ```
/// ```
fn _foo() {}
```

I run the following command:

```
$ cargo test -- -Z unstable-options --format=json
```

This produces the following output:

```
    Finished test [unoptimized + debuginfo] target(s) in 0.01s
     Running target\debug\deps\untitled-d6b954951d0421be.exe
{ "type": "suite", "event": "started", "test_count": 0 }
{ "type": "suite", "event": "ok", "passed": 0, "failed": 0, "allowed_fail": 0, "ignored": 0, "measured": 0, "filtered_out": 0 }
   Doc-tests untitled
{ "type": "suite", "event": "started", "test_count": 1 }
{ "type": "test", "event": "started", "name": "src\lib.rs - _foo (line 1)" }
{ "type": "test", "name": "src\lib.rs - _foo (line 1)", "event": "ok" }
{ "type": "suite", "event": "ok", "passed": 1, "failed": 0, "allowed_fail": 0, "ignored": 0, "measured": 0, "filtered_out": 0 }
```

For example, this JSON
```
{ "type": "test", "event": "started", "name": "src\lib.rs - _foo (line 1)" }
```
can't be parsed, because contains unescaped `\` character.

### Meta

`rustc --version --verbose`:
```
rustc 1.46.0 (04488afe3 2020-08-24)
binary: rustc
commit-hash: 04488afe34512aa4c33566eb16d8c912a3ae04f9
commit-date: 2020-08-24
host: x86_64-pc-windows-msvc
release: 1.46.0
LLVM version: 10.0
```

**Comment by mchernyavsky** [comment]
P.S. - On Unix, I also can create a directory with backslash in name (e.g. `mkdir 'qwe\qwe'`).

**Comment by matklad** [comment]
Relevant code is here: https://github.com/rust-lang/rust/blob/8bfe289886af727150c1b9ec502cbfd7bbf425e8/library/test/src/formatters/json.rs

Not sure why this doesn't use serde/rustc-sereialize, but, if the reason is "we want to be as simple as possible", I have a super-simple library for outputting JSON here: https://github.com/matklad/write-json/

**Comment by euclio** [comment]
libtest can't easily depend on crates: they need the "rustc-dep-of-std" hack. That's probably why serde wasn't used here.

**Comment by ehuss** [comment]
Closing as I believe this was fixed by #77890.
